media_driver/agnostic/gen11/codec/kernel_free/media_srcs.cmake

```diff
@@ -18,6 +18,35 @@
 # ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 # OTHER DEALINGS IN THE SOFTWARE.
 
+if (BUILD_KERNELS)
+
+    function(gen_cm_kernel_from_source name platform)
+
+        set(cm_krn_dir ${CMAKE_SOURCE_DIR}/media_driver/agnostic/${platform}/codec/kernel_free)
+        set(cm_krn ig${name}krn_g11)
+
+        # Generating source from the .krn file
+        add_custom_command(
+            OUTPUT ${cm_krn_dir}/${cm_krn}.c ${cm_krn_dir}/${cm_krn}.h
+            DEPENDS KernelBinToSource
+            COMMAND ${CMAKE_COMMAND} -E rename ${CMAKE_CURRENT_LIST_DIR}/commonkernel.krn ${CMAKE_CURRENT_LIST_DIR}/${cm_krn}.krn
+            COMMAND KernelBinToSource ARGS -i ${CMAKE_CURRENT_LIST_DIR}/${cm_krn}.krn -o ${cm_krn_dir} -v ${cm_krn} -index 14 -t 18
+            COMMAND ${CMAKE_COMMAND} -E remove ${CMAKE_CURRENT_LIST_DIR}/${cm_krn}.krn
+            COMMAND ${CMAKE_COMMAND} -E remove_directory ${CMAKE_CURRENT_LIST_DIR}/Binary
+            COMMENT "Generate source file from krn")
+
+    endfunction()
+
+    if(GEN11_ICLLP)
+
+        exec_program(${PYTHON} ${CMAKE_CURRENT_LIST_DIR}
```

Comment:
In this way you will run this command within configuration stage. I.e. you mix configuration and compilation stages. This will not do. Please, refactor in a way which will just configure targets and build them within make.

Reply:
For media driver, it is not compilation. It is still in configration. if no kernel changes, file will not be chagned.

Reply:
That's exactly the issue: you are running compilation commands within configuration stage. That's wrong.

Reply:
To avoid exec_program you should specify input/output dependencies for the add_custom_command:
* OUTPUT: kernel .h/.c files
* DEPENDENCIES: KernelBinToSource <cm .c files>

In this way make will know that to to produce ihd_drv_video.so it need to produce kernel .h/.c files first and to do that it should run your custom commands, i.e. generate KernlBinToSource application and use .cm files as input dependencies.